crates/metrics/src/compression_metrics.rs

```diff
@@ -0,0 +1,55 @@
+use crate::global_registry;
+use prometheus_client::metrics::gauge::Gauge;
+use std::sync::{
+    atomic::{
+        AtomicU32,
+        AtomicU64,
+    },
+    OnceLock,
+};
+
+#[derive(Debug)]
+pub struct CompressionMetrics {
+    pub compression_ratio: Gauge<f64, AtomicU64>,
+    pub compression_duration_ms: Gauge<u32, AtomicU32>,
+    pub compression_block_height: Gauge<u32, AtomicU32>,
+}
+
+impl Default for CompressionMetrics {
+    fn default() -> Self {
+        let compression_ratio = Gauge::default();
+        let compression_duration_ms = Gauge::default();
+        let compression_block_height = Gauge::default();
+
+        let metrics = CompressionMetrics {
+            compression_ratio,
+            compression_duration_ms,
+            compression_block_height,
+        };
+
+        let mut registry = global_registry().registry.lock();
+        registry.register(
+            "compression_ratio",
+            "Compression ratio",
+            metrics.compression_ratio.clone(),
+        );
+        registry.register(
+            "compression_duration_ms",
+            "Compression duration in milliseconds",
+            metrics.compression_duration_ms.clone(),
+        );
+        registry.register(
+            "compression_block_height",
+            "Compression block height",
+            metrics.compression_block_height.clone(),
+        );
+
```

Comment:
Nit: Not a fan of side-effects in `Default` implementations. Spontaneously I'd prefer to see a derived `Default` implementation, and a separate ".register` method that takes the registry as an input and registers these metrics.

Reply:
this is how it is for all other metrics modules. while the comment is valuable, let's save that refactor for when we're fixing all of them :)

Reply:
Fair enough :+1: